src/pipeline-stages/src/main/python/UDFTransformer.py

```diff
@@ -0,0 +1,139 @@
+from pyspark.ml.common import inherit_doc
+from mmlspark.Utils import *
+
+@inherit_doc
+class UDFTransformer(ComplexParamsMixin, JavaMLReadable, JavaMLWritable, JavaTransformer):
+    """
+
+
+    Args:
+
+        inputCol (str): The name of the input column (default: )
+        outputCol (str): The name of the output column
+        udf (object): User Defined Python Function to be applied to the DF input col
+        udfScala (object): User Defined Function to be applied to the DF input col
+    """
+
+    @keyword_only
+    def __init__(self, inputCol=None, inputCols=None, outputCol=None, udf=None):
+        super(UDFTransformer, self).__init__()
+        self._java_obj = self._new_java_obj("com.microsoft.ml.spark.UDFTransformer")
+        self.inputCol = Param(self, "inputCol", "inputCol: The name of the input column (default: )")
+        self.inputCols = Param(self, "inputCols", "inputCols: The names of the input columns (default: )")
+        self.outputCol = Param(self, "outputCol", "outputCol: The name of the output column")
+        self.udf = Param(self, "udf", "udf: User Defined Python Function to be applied to the DF input col")
+        if udf != None:
+            self.setUDF(udf)
+            self._udf = udf
+        if inputCol != None:
+            self.setInputCol(inputCol)
+        if inputCols != None:
+            self.setInputCols(inputCols)
+        if outputCol != None:
+            self.setOutputCol(outputCol)
+
+    def setInputCol(self, value):
+        """
+
+        Args:
+
+            inputCol (str): The name of the input column (default: )
+
+        """
+        self._set(inputCol=value)
+        return self
+
+
+    def getInputCol(self):
+        """
+
+        Returns:
+
+            str: The name of the input column (default: )
+        """
+        return self.getOrDefault(self.inputCol)
+
+
+    def setInputCols(self, value):
+        """
+
+        Args:
+
+            inputCols (list): The names of the input columns (default: )
+
+        """
+        self._set(inputCols=value)
+        return self
+
+    def getInputCols(self):
+        """
+
+        Returns:
+
+            str: The name of the input column (default: )
+        """
+        return self.getOrDefault(self.inputCols)
+
+    def setOutputCol(self, value):
+        """
+
+        Args:
+
+            outputCol (str): The name of the output column
+
+        """
+        self._set(outputCol=value)
+        return self
+
+
+    def getOutputCol(self):
+        """
+
+        Returns:
+
+            str: The name of the output column
+        """
+        return self.getOrDefault(self.outputCol)
+
+    def setUDF(self, udf):
+        func = None if not hasattr(udf, "func") else udf.func
+        returnType = None if not hasattr(udf, "returnType") else udf.returnType
+        userDefinedFunction = UserDefinedFunction(func, returnType = returnType, name = __name__)
+        self._java_obj = self._java_obj.setUDF(userDefinedFunction._judf)
```

Comment:
Please just pass the UDF through to the `UserDefinedFunction` constructor. If it fails, that's consistent with Python error semantics..